### PR TITLE
Refactor delegate and internal classes

### PR DESCRIPTION
**Pull request by koji-1009**
- Update demo VC
- create `internal/` directory
- Make errors easier to understand
- Update `StudyplusLoginDelegate`

### CODE REVIEW

File: Lib/StudyplusSDK/Studyplus.swift [hunk 1]
```diff
@@ -165,7 +126,21 @@ final public class Studyplus {
...
         }
 
-        StudyplusAPIRequest(accessToken: accessToken).post(record, completion: completion)
+        StudyplusAPI(accessToken: accessToken).post(record, completion: { result in
+            switch result {
+            case .failure(let error):
+                switch error {
+                case .loginRequired:
+                    // clear invalid access token
+                    StudyplusKeychain.deleteAll(serviceName: self.serviceName)
```
Comment (m-yamada04) [inline Lib/StudyplusSDK/Studyplus.swift:135]: 👍

**Review by m-yamada04 — APPROVED** [review]
LGTM!

**Review by nappannda — APPROVED** [review]
LGTM〜〜

**Review by koishi — APPROVED** [review]
🙆🏼‍♂️

**Pull request merged**